File: easy/3194_minimum_average_of_smallest_and_largest_elements/solution.py
from typing import List


class Solution:
    def minimumAverage(self, nums: List[int]) -> float:
        nums.sort()

        left = 0
        right = len(nums) - 1

        minAverage = float("inf")

        while left <= right:
            minAverage = min(minAverage, (nums[left] + nums[right]) / 2)
            left += 1
            right -= 1

        return minAverage

File: easy/3194_minimum_average_of_smallest_and_largest_elements/solution.cpp
#include <algorithm>
#include <limits>
#include <ranges>
#include <vector>

class Solution {
public:
  double minimumAverage(std::vector<int> &nums) {
    std::ranges::sort(nums);

    size_t left = 0;
    size_t right = nums.size() - 1;

    double minAverage = std::numeric_limits<double>::max();

    while (left <= right)
      minAverage = std::min(
          minAverage, static_cast<double>(nums[left++] + nums[right--]) / 2.0);

    return minAverage;
  }
};
